### PR TITLE
Configurable expression languages

### PR DESCRIPTION
**Pull request by CarstenWickner**
The intention here is to introduce four new configuration options:
1. Setting the default Expression Language for Headers (and Output Cells) -- thereby also fixing the currently wrong default mentioned in #405.
2. Setting the default Expression Language for Input Cells -- just to be consistent with the above and avoid it being hard-coded in various places.
3. Setting the available Expression Language Options for Headers and Output Cells -- thereby allowing to hide unwanted options (e.g. if there is no Groovy or JRuby run-time where the decision table will be evaluated or FEEL just doesn't make sense) and add additional ones if desired.
4. Setting the available Expression Language Options for Input Cells -- same as point 3 but to enable different options (e.g. to only offer FEEL here).

The first two points are effectively #19.
The latter two points were already mentioned in my request from three years ago (for the non-react modeler):
> 7. Limit the available script languages for input expressions (to match the ones we actually provide on the classpath).
- https://forum.bpmn.io/t/customization-options-for-dmn-js-modeler/867

------

Related to #19 
Related to #405

### CODE REVIEW

File: packages/dmn-js-decision-table/src/features/decision-table-head/editor/components/InputCellContextMenu.js [hunk 1]
```diff
@@ -83,6 +83,8 @@ export default class InputCellContextMenu extends Component {
     return (
       <div className="context-menu-container input-edit">
         <InputEditor
+          defaultExpressionLanguage={ this.getValue('defaultInputExpressionLanguage') }
+          expressionLanguageOptions={ this.getValue('inputExpressionLanguageOptions') }
```
Comment (CarstenWickner) [inline packages/dmn-js-decision-table/src/features/decision-table-head/editor/components/InputCellContextMenu.js]: This is one of the places where I struggle to understand how to pass those values through from the top-level modeler component.

File: packages/dmn-js-decision-table/src/features/decision-table-head/editor/components/InputEditor.js [hunk 2]
```diff
@@ -6,35 +6,25 @@ import ContentEditable from 'dmn-js-shared/lib/components/ContentEditable';
...
     this.setExpressionLanguage = (expressionLanguage) => {
       this.handleChange({ expressionLanguage });
     };
 
-    this.makeScript = (event) => {
-      event.preventDefault();
-      event.stopPropagation();
-
-      this.setExpressionLanguage('FEEL');
-    };
```
Comment (CarstenWickner) [inline packages/dmn-js-decision-table/src/features/decision-table-head/editor/components/InputEditor.js]: To simplify this a bit here, I've discarded the one-line input field and would show the multi-line editor every time.
As configuration user, I always find it annoying being forced to do that extra click for complex expressions and for simple ones I don't mind having a bit of empty space below either way.

Can of course revert this part, as it is not really necessary for making the expression languages configurable.

File: packages/dmn-js-decision-table/src/features/expression-language/ExpressionLanguage.js [hunk 3]
```diff
@@ -44,8 +27,25 @@ export default class ExpressionLanguage {
...
         }
 
+        // this needs to come from somewhere
+        const configOptions = {};
+        const {
+          defaultInputExpressionLanguage = 'FEEL',
+          defaultHeaderAndOutputExpressisonLanguage = 'JUEL',
+          inputExpressionLanguageOptions = DEFAULT_INPUT_EXPRESSION_LANGUAGE_OPTIONS,
+          headerAndOutputExpressionLanguageOptions = DEFAULT_HEADER_AND_OUTPUT_EXPRESSION_LANGUAGE_OPTIONS
+        } = configOptions;
```
Comment (CarstenWickner) [inline packages/dmn-js-decision-table/src/features/expression-language/ExpressionLanguage.js]: One more place I'd need a pointer as to where those need to be passed-through.

**Comment by barmac** [comment]
Hi @CarstenWickner and thank you for your contribution. I modified a bit your first commit on [a new branch](https://github.com/bpmn-io/dmn-js/tree/418-JUEL_as_default_header_expression_language) so that the CI can build it and there are no misleading labels. I also adjusted the commit message to adhere to our [commit message guidlines](https://github.com/bpmn-io/dmn-js/blob/develop/docs/project/COMMIT_MESSAGES.md). After the weekend, we can continue with the configurability feature.

**Comment by pinussilvestrus** [comment]
Thanks Maciej!

**Comment by barmac** [comment]
@CarstenWickner I merged the change of the default expression language via https://github.com/bpmn-io/dmn-js/commit/6a80e3afad1c239215b4a6cf9490a821967ffab6.

**Comment by barmac** [comment]
Regarding the configurability of the expression languages, this needs yet some more discussion. E.g. it might be worth it to actually configure one expression language for a whole column instead of separate cells. Please feel free to create an issue in which the feature might be discussed and then we can continue to work on this.

**Comment by barmac** [comment]
Thanks for creating the issue. I'll get back to this PR later this or next week so that we can ship the actual feature.

**Comment by barmac** [comment]
I'm closing this in favor of https://github.com/bpmn-io/dmn-js/pull/441

@CarstenWickner Thanks for your contribution!

**Comment by nikku** [comment]
Merged via https://github.com/bpmn-io/dmn-js/pull/441. 

Thanks for contributing your initial version :+1:

**Pull request closed without merging**